sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -511,6 +507,208 @@ class ConditionalHashJoinIterator(
+      }.getOrElse {
+        unconditionalLeftJoinGatherMaps(leftKeys, rightKeys)
+      }
+      assert(maps.length == 2)
+      try {
+        val lazyLeftMap = LazySpillableGatherMap(maps(0), spillCallback, "left_map")
+        val lazyRightMap = LazySpillableGatherMap(maps(1), spillCallback, "right_map")
+        val gatherer = JoinGatherer(lazyLeftMap, leftData, lazyRightMap, rightData,
+          OutOfBoundsPolicy.DONT_CHECK, OutOfBoundsPolicy.NULLIFY)
+        if (gatherer.isDone) {
+          // Nothing matched...
+          gatherer.close()
+          None
+        } else {
+          updateTrackingMask(lazyRightMap)
+          Some(gatherer)
+        }
+      } finally {
+        maps.foreach(_.close())
+      }
+    }
+  }
+
+  override def close(): Unit = {
+    if (!closed) {
+      super.close()
+      compiledConditionRes.foreach(_.close())
+      rightTracker.foreach(_.close())
+    }
+  }
+
+  override def hasFinalBatch(): Option[ColumnarBatch] = {
+    rightTracker match {
+      case None => None
+      case Some(tracker) => {
+        val filteredBatch = withResource(tracker.releaseBatch()) { trackerBatch =>
+          withResource(GpuColumnVector.from(trackerBatch)) { trackerTab =>
+            val batch = built.getBatch
+            withResource(GpuColumnVector.from(batch)) { builtTable =>
+              withResource(builtTable.filter(trackerTab.getColumn(0))) { filterTab =>
+                GpuColumnVector.from(filterTab, GpuColumnVector.extractTypes(batch))
+              }
+            }
+          }
+        }
+        // Combine right columns with null columns for left side
+        val ret = withResource(filteredBatch) { rightBatch =>
+          val numFilterRows = rightBatch.numRows()
+          if (numFilterRows > 0) {
+            val leftColumns = streamAttributes.map { attr =>
+              GpuColumnVector.fromNull(numFilterRows, attr.dataType)
+            }
+            withResource(new ColumnarBatch(leftColumns.toArray, numFilterRows)) { leftBatch =>
+              Some(GpuColumnVector.combineColumns(leftBatch, rightBatch))
+            }
+          } else {
+            None
+          }
+        }
+        rightTracker = None
+        ret
+      }
+    }
+  }
+
+  private def trueColumnTable(numRows: Int): Table = {
+    withResource(Scalar.fromBool(true)) { trueScalar =>
+      withResource(ai.rapids.cudf.ColumnVector.fromScalar(trueScalar, numRows)) {
+        new Table(_)
+      }
+    }
+  }
+
+  // Create a boolean column to indicate which gather map rows are valid.
+  private def validIndexMask(gatherView: ColumnView): ColumnView = {
+    withResource(Scalar.fromInt(-numRightRows)) { minValue =>
+      withResource(Scalar.fromInt(numRightRows)) { maxValue =>
+        withResource(gatherView.greaterOrEqualTo(minValue)) { greaterEqualMin =>
+          withResource(gatherView.lessThan(maxValue)) { lessMax =>
+            lessMax.and(greaterEqualMin)
+          }
+        }
+      }
+    }
```

Comment:
This can be simplified to a check for != Int.MinValue. cudf should not be generating any value for a gather map that is out of bounds except for Int.MinValue.

Reply:
Thanks!  That is much simpler.

Reply:
I made this change.